cdap-ui/app/cdap/components/FastAction/FastActionButton/index.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {PropTypes} from 'react';
+
+export default function FastActionButton({icon, api, params, onSuccess, onError, onClick}) {
+  function clickHandler() {
+    if (typeof onClick === 'function') {
+      onClick();
+    }
+
+    api(params)
+      .subscribe(onSuccess, onError);
+  }
+
+  return (
+    <button
+      className="btn btn-link"
+      onClick={clickHandler}
+    >
+      <span className={icon}></span>
+    </button>
+  );
+}
+
+
+FastActionButton.propTypes = {
+  icon: PropTypes.string,
+  api: PropTypes.func,
+  params: PropTypes.object,
```

Comment:
Hhmm the purpose of prop validation is to give the component entire control of what it gets. If thats not the case then we need to think about how to define the contract. We can define all possible params we might get for each fast action and use `oneOf` operator for the prop.